### PR TITLE
chore(cli-repl): MONGOSH-377 separate out autocompleter into its own lerna package

### PR DESCRIPTION
**Pull request by lrlna**
## Description

This PR creates a new `lerna` package inside this repository for autocompletion.

replaces #200

### CODE REVIEW

File: packages/autocomplete/README.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+const line = 'db.coll.re';
+const completions = autocomplete(serverVersion, line);
+// returns:
+// [
+//   [ 'db.coll.renameCollection', 'db.coll.replaceOne', 'db.coll.reIndex' ],
+//   'db.coll.re' 
+// ]
+```
+
+Autocomlete is based on currently implemeted APIs in [@mongosh/shell-api](https://www.npmjs.com/package/@mongosh/shell-api)
```
Comment (addaleax) [inline packages/autocomplete/README.md:42]:
```suggestion
Autocomplete is based on currently implemeted APIs in [@mongosh/shell-api](https://www.npmjs.com/package/@mongosh/shell-api)
```

File: packages/autocomplete/index.spec.ts [hunk 2]
```diff
@@ -72,7 +72,7 @@ describe('completer.completer', () => {
     });
 
     it('is an exact match to one of shell completions', () => {
-      const i = 'db.bios.find({ field: { $exis';
+      const i = 'db.bios.find({ field: { $exists';
```
Comment (addaleax) [inline packages/autocomplete/index.spec.ts:75]: I might be misunderstanding this, but wouldn’t the point of this test be that it completes `$exis` to `$exists`?

**Review by addaleax — APPROVED** [review]

File: packages/autocomplete/index.spec.ts (hunk 2 above)
Comment (lrlna) [reply packages/autocomplete/index.spec.ts:75]: lol yep, you're right!

File: packages/autocomplete/.eslintignore [hunk 3]
```diff
@@ -0,0 +1,3 @@
+node_modules
+local-test.js
```
Comment (mcasimir) [inline packages/autocomplete/.eslintignore:2]: is this a file we commit?

**Review by mcasimir — COMMENTED** [review]
nice!

File: packages/autocomplete/.eslintignore (hunk 3 above)
Comment (lrlna) [reply packages/autocomplete/.eslintignore]: nope! thanks for the spot

**Pull request merged**